### PR TITLE
fix: don't log certain fields

### PR DESCRIPTION
**Pull request by kshamajain99**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: util/exec/exec.go [hunk 1]
```diff
@@ -36,11 +37,24 @@ func Run(cmd *exec.Cmd) (string, error) {
 func RunWithRedactor(cmd *exec.Cmd, redactor func(text string) string) (string, error) {
 	span := tracing.NewLoggingTracer(log.NewLogrusLogger(logrus.New())).StartSpan(fmt.Sprintf("exec %v", cmd.Args[0]))
 	span.SetBaggageItem("dir", fmt.Sprintf("%v", cmd.Dir))
-	span.SetBaggageItem("args", fmt.Sprintf("%v", cmd.Args))
+
+	var args []string
```
Comment (kshamajain99) [inline util/exec/exec.go:41]: @alexmt This is a temporary solution, I am going to fix it in gitops-engine/NewLoggingTracer by adding redactor.

File: util/exec/exec.go [hunk 2]
```diff
@@ -36,7 +37,8 @@ func Run(cmd *exec.Cmd) (string, error) {
 func RunWithRedactor(cmd *exec.Cmd, redactor func(text string) string) (string, error) {
 	span := tracing.NewLoggingTracer(log.NewLogrusLogger(logrus.New())).StartSpan(fmt.Sprintf("exec %v", cmd.Args[0]))
 	span.SetBaggageItem("dir", fmt.Sprintf("%v", cmd.Dir))
-	span.SetBaggageItem("args", fmt.Sprintf("%v", cmd.Args))
+	args := regexp.MustCompile("(--username|--password) [^ ]*").ReplaceAllString(fmt.Sprintf("%v", cmd.Args), "$1 ******")
```
Comment (alexmt) [inline util/exec/exec.go:40]: I think `redactor` parameter already knows how to reduce sensitive data. So the fix is rather 👍 
```
span.SetBaggageItem("args", redactor(fmt.Sprintf("%v", cmd.Args)))
```

**Review by alexmt — CHANGES_REQUESTED** [review]

File: util/exec/exec.go (hunk 2 above)
Comment (kshamajain99) [reply util/exec/exec.go]: Yes.

**Review by alexmt — APPROVED** [review]
LGTM

**Comment by kshamajain99** [comment]
https://github.com/argoproj/gitops-engine/issues/232

**Pull request merged**